### ISSUE TITLE
Create nginx hosts only for "running" container

### ISSUE THREAD

**Issue by rtyshyk** [adi90x/rancher-active-proxy#51]
Hi, 

Thanks for a great tool.

I noticed that there are no checks for container state.

As result container `upstream` may contain two IP addresses when the upgrade is not approved (old container still alive), or even IP part is missing on `upstream` when config generated at the moment when the new container is starting.

Missing IP address causing invalid nginx conf as well as a duplicate entry for the same container with the right IP address (on next config update)

#I might be mistaken somewhere, but in general, I have one-two times for a week this issue, it is more or less reproducible with heavy container where container entrypoint takes some time to start.

**Comment by adi90x** [comment]
Hello,
Yeah I 'm aware of that kind of situation to avoid that we need to find a
way to check and keep only active containers !
Will try to find something for that !

**Comment by rtyshyk** [comment]
go-rancher-gen has "state" and "health" properties for a container, so the one more if statement can be added to a dynamic container configuration

**Comment by adi90x** [comment]
Ok was not sure if the property was already available in rancher-gen !
It is quite easy then, will try to push something this afternoon

**Comment by adi90x** [comment]
Looking at actual code it seems that there is already a check and it should
only get healthy and running containers...
Look at line 316, I will try to look again into that (it was merged from a
PR some time ago)

Le 6 mars 2018 13:02, "Adrien Maurel" <amaurel90@gmail.com> a écrit :

Ok was not sure if the property was already available in rancher-gen !
It is quite easy then, will try to push something this afternoon

**Comment by adi90x** [comment]
Just made some more test, it seems to work as expected... While upgrading
it only show working and healthy containers...
Do you have some logs  ?
Are you sure that your are running the last RAP version ?

**Comment by rtyshyk** [comment]
Sorry, my bad, I did not find any related issue in the changelog from version 0.9.2 which I use.
Will try to update and let you know.
Actually, I used a fork from v0.9.2 with some custom ldap and so on.

**Comment by adi90x** [comment]
Is it a public fork ?
We may pull some commit from it

**Comment by rtyshyk** [comment]
It is not public.

We use it for our dev infrastructure. I added next features:
- basic auth based on ldap (https://github.com/kvspb/nginx-auth-ldap), ldap might be enable skipped for all RAP instance or for the specific container
- "secret link" to bypass ldap authentication  (e.g. to share something who has no ldap account)
- bypass ldap basic auth if request made from docker local network

Is it interesting to you?

**Comment by adi90x** [comment]
Yeah the online update was merged between v0.9.2 and v0.9.3  (almost a year
happen between) and I merged quite a few PR !
Need to find a way to auto tag/ auto update version  !

**Comment by adi90x** [comment]
Yeah if it is not too critical, would appreciate if you open a PR for that !
Quite sure some other people would be interested too

**Comment by rtyshyk** [comment]
I merged the latest change to my fork. looks ok. 

Should I create PR on gitlab or github?

**Comment by adi90x** [comment]
Gitlab.com is easier for me to merge but if you are more used to Github.com
will move it !

**Comment by adi90x** [comment]
Closing this issue , [https://gitlab.com/adi90x/rancher-active-proxy/merge_requests/32](PR posted on GitLab) , do not hesitate if you have any other issue !

**Comment by rtyshyk** [comment]
The issue still reproducible (from time to time).

I have two times completely the same upstream + hosts configuration ({{ define "server" }} section).

There are even the same IP addresses in two templates.
old container IP address is $alive, and the old one is not (commented)

I completely out of ideas how it is possible :( 
Do you have some?
